solvebio/query.py

```diff
@@ -880,26 +884,31 @@ def join(self, query_b, key, key_b=None, prefix="b_", always_prefix=True):
             else:
                 name = field.name
 
+            query_b_prefixed_fields.append(name)
+
             if name in existing_field_names:
-                logger.warning("Field '{}' will be overwritten in the join results")
+                logger.warning("Field '{}' will be overwritten in the join results".format(name))
 
             target_fields.append({
                 "name": name,
                 "title": field.title,
                 "data_type": field.data_type,
                 "ordering": field.ordering,
-                "is_list": False,
+                "is_list": True,
                 "is_transient": False,
                 "expression": """
-                    get(record, "{}.{}")
-                """.format(query_b_join_field_name, field.name),
+                    [get(item, "{}") for item in record.{}]
+                """.format(field.name, query_b_join_field_name),
                 "depends_on": [query_b_join_field_name]
             })
 
         # Add to any existing target fields
         new_query._target_fields += target_fields
 
-        return new_query
+        # Explode new_query records
+        return new_query.annotate(
```

Comment:
What this does is, it will submit the first query and then for each chunk of records, it will issue another set of API requests to annotate them.

Instead of these extra requests, what you can try is:

```
new_query._annotator_params = {"post_annotation_expression": "explode..."}
```

This will run the post_annotation_expression right in the same API request that the query is running.

Reply:
Then you can just return new_query

Reply:
I tried that, but it wasn't give the right result... Let me check once again.